nav2_theta_star_planner/test/test_theta_star.cpp

```diff
@@ -0,0 +1,163 @@
+
+  void uaddIndex(const int & cx, const int & cy, const int & id_this) {addIndex(cx, cy, id_this);}
+
+  void ugetIndex(const int & cx, const int & cy, int & id_this) {getIndex(cx, cy, id_this);}
+
+  bool runAlgo(std::vector<coordsW> & path)
+  {
+    if (!isSafeToPlan()) {
+      return generatePath(path);
+    }
+    return false;
+  }
+};
+
+init_rclcpp node;
+auto planner_ = std::make_unique<test_theta_star>();
+
+// Tests meant to test the algorithm itself and its helper functions
+TEST(ThetaStarTest, test_theta_star)
+{
+  planner_->costmap_ = new nav2_costmap_2d::Costmap2D(50, 50, 1.0, 0.0, 0.0, 0);
+  for (int i = 20; i <= 30; i++) {
+    for (int j = 20; j <= 30; j++) {
+      planner_->costmap_->setCost(i, j, 253);
+    }
+  }
+
+  planner_->size_x_ = 50;
+  planner_->size_y_ = 50;
+  geometry_msgs::msg::PoseStamped start, goal;
+  start.pose.position.x = 10;
+  start.pose.position.y = 10;
+  start.pose.orientation.w = 1.0;
+  goal.pose.position.x = 40;
+  goal.pose.position.y = 35;
+  goal.pose.orientation.w = 1.0;
+  /// Check if the setStartAndGoal function works properly
+  planner_->setStartAndGoal(start, goal);
+  EXPECT_TRUE(planner_->src_.x == 10 && planner_->src_.y == 10);
+  EXPECT_TRUE(planner_->dst_.x == 40 && planner_->dst_.y == 35);
+
+  /// Check if the initializePosn function works properly
+  planner_->uinitializePosn(50 * 50);
+  EXPECT_EQ(planner_->getSizeOfNodePosition(), (50 * 50)) << "passed this 1";
+
+  /// Check if the withinLimits function works properly
+  EXPECT_TRUE(planner_->uwithinLimits(20, 40));
+  EXPECT_FALSE(planner_->uwithinLimits(120, 140));
+
+  tree_node n = {40, 35, 120, 0, 1, false, 20};
+  /// Check if the isGoal function works properly
+  EXPECT_TRUE(planner_->uisGoal(n));       // both (x,y) are the goal coordinates
+  n.x = 25;
+  EXPECT_FALSE(planner_->uisGoal(n));      // only y coordinate matches with that of goal
+  n.x = 40;
+  n.y = 20;
+  EXPECT_FALSE(planner_->uisGoal(n));      // only x coordinate matches with that of goal
+  n.x = 30;
+  EXPECT_FALSE(planner_->uisGoal(n));      // both (x, y) are different from the goal coordinate
+
+  /// Check if the isSafe functions work properly
+  EXPECT_TRUE(planner_->isSafe(10, 30));       // cost at this point is 0
+  EXPECT_FALSE(planner_->isSafe(25, 25));      // cost at this point is 253 (>LETHAL_COST)
+
+  int c_id;
+  /// Check if the addIndex & getIndex work properly
+  coordsM c = {20, 30};
+  planner_->uaddIndex(c.x, c.y, 0);
+  planner_->ugetIndex(c.x, c.y, c_id);
+  EXPECT_EQ(c_id, 0);
+
+  double sl_cost = 0.0;
+  /// Checking for the case where the losCheck should return the value as true
+  EXPECT_TRUE(planner_->ulosCheck(10, 10, 40, 10, sl_cost));
+  /// and as false
+  EXPECT_FALSE(planner_->ulosCheck(10, 25, 30, 25, sl_cost));
+
+  std::vector<coordsW> path;
+  /// Check if the planner returns a path for the case where a path exists
+  EXPECT_TRUE(planner_->runAlgo(path));
```

Comment:
You should check the size of the path to make sure its correct, not that it just returned true

Reply:
Okay, I am on it!